FitTracker/app/src/main/java/com/pps/globant/fittracker/utils/FacebookLoginProvider.java

```diff
@@ -1,112 +1,122 @@
-                                    });
-                            Bundle parameters = new Bundle();
-                            parameters.putString(FIELDS, REQUESTED_FIELDS);
-                            request.setParameters(parameters);
-                            request.executeAsync();
-                        }
-
-                        @Override
-                        public void onCancel() {
-                        }
-
-                        @Override
-                        public void onError(FacebookException exception) {
-                            bus.post(new FetchingFbUserDataErrorEvent(exception));
-                        }
-                    });
-        }
+    private static final List<String> LOGIN_PERMISSIONS = Collections.singletonList("public_profile");
+    private final Bus bus;
+    private CallbackManager callbackManager;
+    private boolean callbackRegistered;
+
+
+    private void registerCallback() {
+        LoginManager.getInstance().registerCallback(callbackManager,
+                new FacebookCallback<LoginResult>() {
+                    @Override
+                    public void onSuccess(LoginResult loginResult) {
+                        getUserProfile(loginResult.getAccessToken());
+                    }
+
+                    @Override
+                    public void onCancel() {
+                        bus.post(new FetchingFbUserDataCancelEvent());
+                    }
+
+                    @Override
+                    public void onError(FacebookException exception) {
+                        bus.post(new FetchingFbUserDataErrorEvent(exception));
+                    }
+                });
         callbackRegistered = true;
     }
 
-    private static void setResponse(JSONObject jsonObject, Bus bus) {
-        String name;
-        try {
-            name = jsonObject.getString(NAME);
-        } catch (JSONException e) {
-            bus.post(new CantRetrieveAllTheFieldsRequestedsEvent());
-            return;
-        }
-        bus.post(new FetchingFbUserDataCompletedEvent(new FbUser(name)));
+    public FacebookLoginProvider(Bus bus, CallbackManager callbackManager) {
+        this.bus = bus;
+        this.callbackManager = callbackManager;
+        callbackRegistered = false;
     }
 
-    public static void logOut() {
+    public void logOut() {
         LoginManager.getInstance().logOut();
+        bus.post(new LogOutCompleteEvent());
     }
 
-    private static boolean isCallbackRegistered() {
-        return callbackRegistered;
+    public void logIn(Activity activity) {
+        if (!callbackRegistered) {
+            registerCallback();
+        }
+        LoginManager.getInstance().logInWithReadPermissions(activity, LOGIN_PERMISSIONS);
     }
 
-    public static class CantRetrieveAllTheFieldsRequestedsEvent {
-
+    public boolean isLoginTokenActive() {
+        return (AccessToken.getCurrentAccessToken() != null && !AccessToken.getCurrentAccessToken().isExpired());
```

Comment:
You could change by
 
      public boolean isLoginTokenActive() {
          final AccesToken accessToken = AccessToken.getCurrentAccessToken();
            return (accessToken != null && !accessToken.isExpired());
     }

Reply:
done